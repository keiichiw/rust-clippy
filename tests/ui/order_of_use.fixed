// edition:2018
// run-rustfix
// aux-build:order_of_use_helper.rs

#![warn(clippy::order_of_use)]
#![allow(unused)]
#![allow(unused_imports)]

use crate::myself::*;

use crate::myself::*;

use crate::myself::*;

use order_of_use_helper::a::{b1::{c2, *}, b2::c1};

use order_of_use_helper::a::{b1::{c2, *}, b2::c1};

use order_of_use_helper::a::{b1::{c2, *}, b2::c1};

use std::io;

use std::io;

use std::io;

use {order_of_use_helper::*, std::fs};

use {order_of_use_helper::*, std::fs};

use {order_of_use_helper::*, std::fs};


mod myself {
    pub fn a() {}
    pub fn b() {}
    pub fn c() {}
}

extern crate order_of_use_helper;

fn main() {
    // test code goes here
}
