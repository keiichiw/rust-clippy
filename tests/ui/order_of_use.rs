#![warn(clippy::order_of_use)]

fn main() {
    // test code goes here
}
