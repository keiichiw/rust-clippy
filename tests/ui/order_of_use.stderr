error: Put a block of `use` here
   |
   = note: `-D clippy::order-of-use` implied by `-D warnings`

error: Put a block of `use` here
  --> $DIR/order_of_use.rs:9:1
   |
LL | use crate::myself::*;
   | ^^^^^^^^^^^^^^^^^^^^^ help: try: `use crate::myself::*;`

error: Put a block of `use` here
  --> $DIR/order_of_use.rs:10:1
   |
LL | / use order_of_use_helper::a::{
LL | |     b1::{c2, *},
LL | |     b2::c1,
LL | | };
   | |__^ help: try: `use order_of_use_helper::a::{b1::{c2, *}, b2::c1};`

error: Put a block of `use` here
  --> $DIR/order_of_use.rs:14:1
   |
LL | use std::io;
   | ^^^^^^^^^^^^ help: try: `use std::io;`

error: Put a block of `use` here
  --> $DIR/order_of_use.rs:15:1
   |
LL | use {order_of_use_helper::*, std::fs};
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `use {order_of_use_helper::*, std::fs};`

error: aborting due to 5 previous errors

